### PR TITLE
use pytest env var to get the test file path

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

File: cardano_node_tests/tests/conftest.py [hunk 1]
```diff
@@ -236,17 +236,18 @@ def session_autouse(change_dir: Any, close_dbconn: Any, testenv_setup_teardown:
...
 
     The dir is specific to a single test file.
     """
-    p = (
-        temptools.get_pytest_worker_tmp(tmp_path_factory)
-        .joinpath(helpers.get_id_for_mktemp(this_testfile))
-        .resolve()
-    )
+    # get the current test file
+    file_path = (os.getenv("PYTEST_CURRENT_TEST") or "").split("::")[0].split("/")[-1]
```
Comment (mkoura) [inline cardano_node_tests/tests/conftest.py:245]: Current behavior was to replace dots with underscores, so directory is not mistaken for regular file. So e.g. temp dir for `test_cli.py` becomes `test_cli_py`. Please retain this behavior.

Please use a default directory name (e.g. "unknown") instead of empty string. A temp directory should be created even if there is a corner case where the `PYTEST_CURRENT_TEST` is not set.

File: cardano_node_tests/tests/conftest.py [hunk 2]
```diff
@@ -236,17 +236,18 @@ def session_autouse(change_dir: Any, close_dbconn: Any, testenv_setup_teardown:
...
 @pytest.fixture(scope="module")
-def testfile_temp_dir(tmp_path_factory: TempdirFactory, this_testfile: str) -> Path:
+def testfile_temp_dir(tmp_path_factory: TempdirFactory) -> Path:
     """Return a temporary dir for storing test artifacts.
 
     The dir is specific to a single test file.
     """
-    p = (
-        temptools.get_pytest_worker_tmp(tmp_path_factory)
-        .joinpath(helpers.get_id_for_mktemp(this_testfile))
```
Comment (mkoura) [inline cardano_node_tests/tests/conftest.py:246]: This was the only use of `helpers.get_id_for_mktemp` function, so it can be removed from `helpers` module now.

**Review by mkoura — CHANGES_REQUESTED** [review]

File: cardano_node_tests/tests/conftest.py [hunk 3]
```diff
@@ -236,17 +236,23 @@ def session_autouse(change_dir: Any, close_dbconn: Any, testenv_setup_teardown:
...
-        .resolve()
+    # get the current test file
+    file_path = (
```
Comment (mkoura) [inline cardano_node_tests/tests/conftest.py:245]: One small thing - please update the comment so it reflects the fact that the result is a directory name derived from current test file name, + change the variable name (to smtg like `tempdir_name`).

**Review by mkoura — CHANGES_REQUESTED** [review]

**Comment by mkoura** [comment]
@saratomaz that's great idea to use `PYTEST_CURRENT_TEST` env variable, this will make our test files cleaner :+1:
